libmobilecoin/src/transaction.rs

```diff
@@ -1139,6 +1192,352 @@ pub extern "C" fn mc_memo_sender_with_payment_request_memo_get_payment_request_i
     })
 }
 
+/* ==== GiftCodeMemoBuilders ==== */
+
+/// # Preconditions
+///
+/// * `gift_code_funding_note` - must be a null-terminated C string containing
+/// up to 54 valid UTF-8 bytes. The actual note stored on chain is up to 53 null
+/// terminated UTF-8 bytes unless the note is exactly 53 utf-8 bytes long,
+/// in which case, no null bytes are stored. If the C string passed here is
+/// exactly 54 bytes, the last byte MUST be null and that byte will be
+/// removed prior to storage on chain.
+#[no_mangle]
+pub extern "C" fn mc_memo_builder_gift_code_funding_create(
+    gift_code_funding_note: FfiStr,
+) -> FfiOptOwnedPtr<McTxOutMemoBuilder> {
+    ffi_boundary(|| {
+        let note = <&str>::try_from_ffi(gift_code_funding_note)
+            .expect("Failed to decode gift code funding note");
+        let gift_code_funding_memo_builder = GiftCodeFundingMemoBuilder::new(note)
+            .expect("Gift code funding note had invalid utf-8 or was more than 53 bytes long");
```

Comment:
I think the utf-8 error would have happened in `let note = <&str>::try_from_ffi` above.
```suggestion
            .expect("Gift code funding note was more than 53 bytes long");
```

Reply:
Indeed!